app/models/rubric_criterion.rb

```diff
@@ -37,6 +38,17 @@ def update_assigned_groups_count
     self.assigned_groups_count = result.uniq.length
   end
 
+  def scale_marks_if_max_mark_changed
+    max_level_mark = self.levels.max_by(&:mark).mark
+    new_max = self.max_mark
+    return if max_level_mark == new_max
```

Comment:
Rather than doing this, just check if `self`'s `max_mark` attribute was changed. ActiveModel can actually do this (even after saving), check out https://api.rubyonrails.org/classes/ActiveModel/Dirty.html.